src/mongocxx/client.cpp

```diff
@@ -55,6 +55,16 @@ void* client::implementation() const {
     return _impl->client_t;
 }
 
+void client::read_concern(class read_concern rc) {
+    libmongoc::client_set_read_concern(_impl->client_t, rc._impl->read_concern_t);
+}
+
+class read_concern client::read_concern() const {
+    auto rc = libmongoc::client_get_read_concern(_impl->client_t);
+    return (class read_concern){
```

Comment:
^ same question for `write_concern` and `read_preference`.

The good thing about not cacheing is that we can return an owned value to the caller. If we cached it, I think  it would have to be a view.